linuxrc_hostname.md

```diff
@@ -1,31 +1,76 @@
 # hostname setting in installation/rescue system
 
-There are 3 cases:
+The hostname can be set (in order of precedence):
 
-1. no hostname is set
+1. via `hostname` boot option
+2. via DHCP (either from HOSTNAME or FQDN)
+3. Default hostname set by Yast
 
-2. hostname is set via dhcp (either from HOSTNAME or FQDN)
+linuxrc only writes `/etc/install.inf`.
+After that, Yast takes over.
+see https://github.com/yast/yast-network/blob/master/doc/hostname.md
 
-3. hostname is set via `hostname` boot option
+## installation system
 
-linuxrc defaults to `install`, which may be overridden by dhcp (2) or user
-(3). (3) takes precedence over (2).
+- `etc/hostname` is an empty file at the beginning of the installation.
+- `linuxrc` doesn't write anything to `/etc/hostname`.
+- `linuxrc` writes by default `/etc/install.inf::SetHostname=1`.
 
-## installation system
+## if `hostname` boot option is used
+
+1. linuxrc takes the value of `hostname` boot option and writes it to
+   `/etc/install.inf::Hostname`.
+2. linuxrc writes `/etc/install.inf::SetHostnameUsed=1`
+
+#### Example
+
+`hostname=myhost` is used
+
+**/etc/install.inf**
+```
+[...]
+SetHostname: 1
+SetHostnameUsed: 1
+Hostname: myhost
+[...]
+```
+
+## If `hostname` boot option is not used
+
```

Comment:
`hostname install`

Reply:
So, linuxrc always sets the transient hostname before Yast in the installation environment?

Reply:
yes.